src/ghost/ghost-post.service.ts

```diff
@@ -113,4 +129,57 @@ export class GhostPostService {
 
         return await this.postService.deleteByApId(apId, account);
     }
+
+    async createGhostPost(
+        account: Account,
+        data: GhostPost,
+    ): Promise<Result<Post, GhostPostError>> {
+        const existingPost = await this.getApIdForGhostPost(data.uuid);
+
+        if (existingPost) {
+            return error('post-already-exists');
+        }
+
+        const postResult = await this.postService.handleIncomingGhostPost(
+            account,
+            data,
+        );
+        if (isError(postResult)) {
+            return postResult;
+        }
+
+        const post = getValue(postResult);
+        await this.insertGhostPostMapping(data.uuid, post.apId);
+
+        return ok(post);
+    }
+
+    private async insertGhostPostMapping(ghostUuid: string, apId: URL) {
```

Comment:
It looks like this and `getApIdForGhostPost` is only used once, it maybe better to just inline these in `createGhostPost` (especially if using a transaction)

Reply:
`getApIdForGhostPost` will be used for getting the post for update and delete. Made `insertGhostPostMapping` inline